pkg/flavor/kubernetes/flavor.go

```diff
@@ -139,31 +140,46 @@ func (flavor *Flavor) LoadNodeInfo(node *model.Node) (string, error) {
 
 	if nodeInfo != nil {
 		log.Infof("Node info %s already known to cluster\n", nodeInfo.ObjectMeta.Name)
-	} else {
-		var iqns []string
-		for i := 0; i < len(node.Iqns); i++ {
-			iqns = append(iqns, *node.Iqns[i])
-		}
-
-		var wwpns []string
-		for i := 0; i < len(node.Wwpns); i++ {
-			wwpns = append(wwpns, *node.Wwpns[i])
+		// make sure the nodeInfo has updated information from the host
+		if !reflect.DeepEqual(nodeInfo.Spec.IQNs, getIqnsFromNode(node)) ||
+			!reflect.DeepEqual(nodeInfo.Spec.WWPNs, getWwpnsFromNode(node)) ||
+			!reflect.DeepEqual(nodeInfo.Spec.Networks, getNetworksFromNode(node)) {
+			nodeInfo.Spec.IQNs = getIqnsFromNode(node)
+			nodeInfo.Spec.Networks = getNetworksFromNode(node)
+			nodeInfo.Spec.WWPNs = getWwpnsFromNode(node)
```

Comment:
I think it looks cleaner to do this in 3 separate if blocks...

iqnsFromNode := getIqnsFromNode(node)
if !reflect.DeepEqual(nodeInfo.Spec.IQNs, iqnsFromNode) {
   node.Spec.IQNs = iqnsFromNode
}
...